### PR TITLE
feat(aks): add microsoft defender support

### PR DESCRIPTION
**Pull request by eyenx**
Changes proposed in the pull request:

Adds support for a `microsoft_defender` block introduced with https://github.com/hashicorp/terraform-provider-azurerm/pull/16218/

Signed-off-by: Toni Tauro <toni.tauro@adfinis.com>

### CODE REVIEW

**Comment by eyenx** [comment]
@lonegunmanb can't run the test anymore. 

Always getting same errors:

```
INFO: Styling Terraform configurations...
INFO: Done!
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.

Error: Unsupported block type

  on main.tf line 5:
   5: moved {

Blocks of type "moved" are not expected here.


Error: Unsupported block type

  on main.tf line 178, in resource "azurerm_kubernetes_cluster" "main":
 178:     precondition {

Blocks of type "precondition" are not expected here.


Error: Unsupported block type

  on main.tf line 182, in resource "azurerm_kubernetes_cluster" "main":
 182:     precondition {

Blocks of type "precondition" are not expected here.


Error: Unsupported argument

  on variables.tf line 57, in variable "agents_pool_name":
  57:   nullable    = false

```

The `terraform-test` image we are using is quite outdated (versoin 0.14.4). Could it be because of that?

**Comment by eyenx** [comment]
I now created my own `mcr.microsI was able to run the tests with it:
oft.com/terraform-test:1.2.5`  by grabbing the github.com/azure/terraform-test repo and adjusting some vars. 

I was able to run the rake build, but not the tests:

```
Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
INFO: Linting Terraform configurations...
INFO: Done!
INFO: Styling Terraform README...
INFO: Done!
INFO: Styling Terraform fixture configurations...
INFO: Done!
go: cannot find main module, but found Gopkg.lock in /go/src/terraform-azurerm-aks
        to create a module there, run:
        go mod init
rake aborted!
ERROR: Go test failed!

/go/src/terraform-azurerm-aks/Rakefile:45:in `block (2 levels) in <top (required)>'
Tasks: TOP => full => e2e => integration:test
(See full trace by running task with --trace)
```

**Comment by lonegunmanb** [comment]
> I now created my own `mcr.microsoft.com/terraform-test:1.2.5` by grabbing the https://github.com/azure/terraform-test repo and adjusting some vars.
> 
> I was able to run the rake build, but not the tests:
> 
> ```
> Terraform has been successfully initialized!
> 
> You may now begin working with Terraform. Try running "terraform plan" to see
> any changes that are required for your infrastructure. All Terraform commands
> should now work.
> 
> If you ever set or change modules or backend configuration for Terraform,
> rerun this command to reinitialize your working directory. If you forget, other
> commands will detect it and remind you to do so if necessary.
> INFO: Linting Terraform configurations...
> INFO: Done!
> INFO: Styling Terraform README...
> INFO: Done!
> INFO: Styling Terraform fixture configurations...
> INFO: Done!
> go: cannot find main module, but found Gopkg.lock in /go/src/terraform-azurerm-aks
>         to create a module there, run:
>         go mod init
> rake aborted!
> ERROR: Go test failed!
> 
> /go/src/terraform-azurerm-aks/Rakefile:45:in `block (2 levels) in <top (required)>'
> Tasks: TOP => full => e2e => integration:test
> (See full trace by running task with --trace)
> ```

Thanks @eyenx for opening this Pr! The dockerfile and docker imaged described in readme file is outdated and no longer been maintained, we're working on a new CI pipeline and we're close to a release (We'll release v6.0.0 with this new CI pipeline). I'm not going to update this `mcr.microsoft.com/terraform-test` image and rake tests since they'll be replaced very soon. If you're interested in this new CI, you can try it on my [personal branch](https://github.com/lonegunmanb/terraform-azurerm-aks/tree/ci), we have instructions in readme file on how to run the pr checks and acc tests for our module. For this pr please give me some time to review and test with new CI on my machine. Once we've solved the issues that block our CI pipeline, we'll introduce this pipeline into some legacy azurerm Terraform modules. Thanks for your understanding.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

File: main.tf [hunk 1]
```diff
@@ -166,6 +166,14 @@ resource "azurerm_kubernetes_cluster" "main" {
     }
   }
 
+  dynamic "microsoft_defender" {
```
[the remark this answers is not in the document]
Comment (eyenx) [reply main.tf:169]: gonna do this ra

File: variables.tf [hunk 2]
```diff
@@ -449,4 +449,11 @@ variable "vnet_subnet_id" {
   type        = string
   description = "(Optional) The ID of a Subnet where the Kubernetes Node Pool should exist. Changing this forces a new resource to be created."
   default     = null
-}
\ No newline at end of file
+}
+
+variable "microsoft_defender_enabled" {
```
[the remark this answers is not in the document]
Comment (eyenx) [reply variables.tf:454]: gonna do this ra

File: main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (eyenx) [reply main.tf]: done

File: variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (eyenx) [reply variables.tf]: done

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

File: main.tf [hunk 3]
```diff
@@ -141,6 +141,13 @@ resource "azurerm_kubernetes_cluster" "main" {
       }
     }
   }
+  dynamic "microsoft_defender" {
+    for_each = var.microsoft_defender_enabled ? ["microsoft_defender"] : []
+
+    content {
+      log_analytics_workspace_id = var.log_analytics_workspace_id == null ? azurerm_log_analytics_workspace.main[0].id : var.log_analytics_workspace.id
```
[the remark this answers is not in the document]
Comment (eyenx) [reply main.tf]: Done. Sry for late reply

**Review by lonegunmanb — APPROVED** [review]
Thanks @eyenx, LGTM

**Pull request merged**